src/email/email.js

```diff
@@ -0,0 +1,120 @@
+const { SMTPServer } = require("smtp-server");
+const { MailParser } = require('mailparser');
+const ParseEmailAddress = require("email-addresses");
+const { Logging } = require('../log');
+
+const log = Logging.get("email");
+
+const roomAlias = function(rcptTo, mxDomain) {
+    let localpartRcptTo;
+    for (let i = 0; i < rcptTo.length; i++) {
+        localpartRcptTo = ParseEmailAddress.parseOneAddress(rcptTo[i].address).local;
+        if ( localpartRcptTo.endsWith(mxDomain)) {
+            log.info("Message destination address:", localpartRcptTo);
+            break;
+        }
+    }
+    return getUserIdorAlias(localpartRcptTo);
+};
+const splitAt = index => x => [x.slice(0, index), x.slice(index+1)];
+
+const getUserIdorAlias = function(localPart) {
+
+    // Received room+<roomalias>_hs
+    if (localPart.startsWith('room+')) {
+        log.info("Message destination is a room");
+        let alias = localPart.substring(localPart.indexOf('+')+1);
+        if (alias.lastIndexOf('_') >= 1) {
+            let res = splitAt(alias.lastIndexOf('_'))(alias);
+            res.unshift("room");
+            return res;
+        }
+      return null;
+    }
+
+    // Received user+<username>_<hs>
+    else if (localPart.startsWith('user+')) {
+        log.info("Message destination is a user");
+        let uname = localPart.substring(localPart.indexOf('+')+1);
+        if (uname.lastIndexOf('_') >= 1) {
+            return splitAt(uname.lastIndexOf('_'))(uname).unshift("user");
+        }
+      return null;
+    }
+
+    log.warn("Destination is not valid");
+    return null;
+};
+
+exports.startSMTP = function (config) {
+    const SMTP = new SMTPServer({
+        secure: false,
+        logger: false,
+        disabledCommands: ['AUTH', 'STARTTLS'],
+        authOptional: true,
+
+        onData: function (stream, session, callback) {
+            let subject, text;
+            const mailparser = new MailParser();
+            const fromAdd = ParseEmailAddress.parseOneAddress(session.envelope.mailFrom.address);
+            let receivedAddress = roomAlias(session.envelope.rcptTo, config.bridge.domain);
+            let alias = "";
+            if (!receivedAddress) {
+                return;
+            }
+            else if (receivedAddress[0] === "room") {
+                alias = `#${receivedAddress[1]}:${receivedAddress[2]}`;
+            }
+            else if (receivedAddress[0] === "user") {
+                // TODO: send message to user
+                return;
+            }
+            mailparser.on('headers', headers => {
+                subject = headers.get('subject');
+            });
+
+            mailparser.on('data', data => {
+                if (data.type === 'text') {
+                    text = data.text;
+                }
+            });
+
+            mailparser.on('end', () => {
```

Comment:
So you've got a few promises in this bit of code which are tricky to handle because the event handler needs a synchronous function. What I would recommend you do is create a function above like `async function handleMail(text, fromAdd, alias)` which you can use async syntax for.

You can then add a simple:
```
mailparser.on('end', () => {
    handleMail(...).catch((err) => log.error(`Could not handle mail:`, err));
});
```